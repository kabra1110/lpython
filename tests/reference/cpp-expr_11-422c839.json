{
    "basename": "cpp-expr_11-422c839",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/../integration_tests/expr_11.py",
    "infile_hash": "d5027dd0f515c653c66202e02cc961bfbe09e628e069ce57b5f5ffe2",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr_11-422c839.stdout",
    "stdout_hash": "ee523e892c8a9e54cc670f9305d4c791e165b79679b71f44e6d9167c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}