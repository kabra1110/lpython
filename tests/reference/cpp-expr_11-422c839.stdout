#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

// Forward declarations
void __main____global_statements();
void f();
namespace {
}

// Implementations
void f()
{
    bool b;
    int32_t i;
    i = 3;
    b = (bool)(i);
    assert (b);
    std::cout << b << std::endl;
}

void __main____global_statements()
{
    f();
}

namespace {

void main2() {
    __main____global_statements();
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
